src/legacy/ui/public/agg_types/controls/drop_partials.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+import { EuiFormRow, EuiIconTip, EuiSwitch } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AggParamEditorProps } from '../../vis/editors/default';
+
+function DropPartialsParamEditor({ agg, aggParam, value, setValue }: AggParamEditorProps<boolean>) {
+  if (agg.params.field.name !== agg.getIndexPattern().timeFieldName) {
+    return null;
+  }
+
+  const content = i18n.translate('common.ui.aggTypes.dropPartialBucketsTooltip', {
+    defaultMessage:
+      "Remove buckets that span time outside the time range so the histogram doesn't start and end with incomplete buckets.",
+  });
+
+  const label = (
+    <>
+      <FormattedMessage
+        id="common.ui.aggTypes.dropPartialBucketsLabel"
+        defaultMessage="Drop partial buckets"
+      />{' '}
+      <EuiIconTip position="right" content={content} type="questionInCircle" aria-label={content} />
+    </>
+  );
+
+  return (
+    <EuiFormRow label={label} fullWidth={true} className="visEditorSidebar__aggParamFormRow">
```

Comment:
Switches should not have their label come from EuiFormRow, instead you need to supply it as the `label` prop to the `EuiSwitch`:

```
<EuiSwitch
  label={label}
  checked={value}
  onChange={ev => setValue(ev.target.checked)}
/>
```

<img width="229" alt="Screen Shot 2019-04-03 at 09 30 24 AM" src="https://user-images.githubusercontent.com/549577/55482953-95ca7c80-55f3-11e9-9dc1-7f43f29858ed.png">

Reply:
Placing this label inside the `EuiSwitch` breaks a `mouseover` event (showing a tooltip) on a `EuiIconTip` comp, which is inside the label.
That happens because of `input` `width` value (the input overlaps the icon):

![swithc_label_hover_issue](https://user-images.githubusercontent.com/31325372/55487808-7e14e780-5637-11e9-8983-503c823b0a4d.png)

And here is a question to you @cchaos and @timroes : Should I override styles for that, or it should be fixed on the EUI side ?

P.S.: definitely I could place the `EuiIconTip` next to `EuiSwitch` :

![image](https://user-images.githubusercontent.com/31325372/55488394-b1a44180-5638-11e9-9ea4-9781cf1e5b24.png)

but I do not think it is a good idea..
Also I didn't find any similar case at [Elastic UI framework](https://elastic.github.io/eui/#/) ...

Reply:
Have you tried just wrapping the whole `EuiSwitch` in a `EuiToolTip`? No need for the small hover area of an icon, just anytime they hover on the switch, it will show up.

Reply:
Can you also show me a screenshot of how:

> Placing this label inside the EuiSwitch breaks a mouseover event (showing a tooltip) on a ... because of input width value (the input overlaps the icon):

Reply:
> Can you also show me a screenshot of how:

the tooltip is not shown at all in this case 😢
and there are also an amount of ways to make it work again using CSS

![drop_partials](https://user-images.githubusercontent.com/31325372/55491069-840dc700-563d-11e9-9b30-31fc9d955f9c.gif)

Reply:
> Have you tried just wrapping the whole `EuiSwitch` in a `EuiToolTip`? No need for the small hover area of an icon, just anytime they hover on the switch, it will show up.

Does it mean that it will be without an `questionInCircle` icon?
That's how it looks like previously:

![image](https://user-images.githubusercontent.com/31325372/55491531-4bbab880-563e-11e9-9580-67638ee6c5b2.png)

Reply:
Ah I see, it's because the hidden input is stretching the entire width even on top of the label and therefore stealing focus from the tooltip. I think this is actually a mistake in EUI and the input should only stretch as wide as the actual switch. We don't need it to cover the label since the label contains a "for" attribute that when clicked will trigger the input selection. 

I will look into a fix on the EUI side since I can foresee this affecting others as well. For now, you could just keep the icon tip separate from the switch as you had here: https://user-images.githubusercontent.com/31325372/55488394-b1a44180-5638-11e9-9ea4-9781cf1e5b24.png

But what styles do you need to play with in that instance?
> but I do not think it is a good idea, since I should also play with styles there..

Reply:
> But what styles do you need to play with in that instance?

@cchaos firstly, as `EuiFormRow` has `flex-direction: column` I should rewrite it `row`. the next step I should style the label, because in common `visEditorSidebar` it has `display: block`.
Also you could see on a screenshot attached below, that `EuiSwitch` has different `font-size` (14px against 12px which all form labels above have), and FMPOV it looks a bit inconsistently:

![image](https://user-images.githubusercontent.com/31325372/55548135-81fa4580-56db-11e9-8105-7b7cb3d0ab1e.png)


All the changes could be find here:
https://github.com/elastic/kibana/blob/b48b5610370fc295af7ae0fe24770e713bd14c5b/src/legacy/ui/public/vis/editors/default/_sidebar.scss#L228

@cchaos I kindly ask you to provide your opinion about it 😊

Reply:
We've talked it over in EUI https://github.com/elastic/eui/issues/1788, specifically this comment: https://github.com/elastic/eui/issues/1788#issuecomment-479907504 and have decided for these condensed/editor style controls, we should just wrap each control in an EuiToolTip. It works for both tabbing and accessibility and limits the noise of these extra icons that aren't needed past the first encounter.

So I'd suggest, remove the `EuiFormRow` altogether, since there's no benefit without labels or help text, and just do something like:

```jsx
<EuiToolTip
  content={content}
  delay="long"
>
  <EuiSwitch
    label={label}
    checked={value}
    onChange={ev => setValue(ev.target.checked)}
  />
</EuiToolTip>
```

Don't override the switch's font styling, we understand that they're not similar to form row labels.